src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,572 @@
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client' . $timestamp,
+                'owner' => 1,
+                'quota' => - 1
+            ]
+        ]);
+        $response = $httpClient->request('GET', '/api/clients/' . $timestamp);
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+    }
+
+    public function testGetItem(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client' . $timestamp,
+                'owner' => 1,
+                'quota' => - 1
+            ]
+        ]);
+        $iri = $this->findIriBy(Client::class, [
+            'name' => 'client' . $timestamp
+        ]);
+        $response = $httpClient->request('GET', $iri);
+
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+            'isActive' => true,
+            'maxParallelJobs' => 1,
+            'name' => 'client' . $timestamp,
+            'owner' => 1,
+            'postScripts' => [],
+            'preScripts' => [],
+            'quota' => '-1',
+            'rsyncLongArgs' => null,
+            'rsyncShortArgs' => null,
+            'sshArgs' => null,
+            'url' => ""
+        ]);
+    }
+
+    public function testUpdateClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'description' => 'description',
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client'.$timestamp.'0',
+                'owner' => 1,
+                'postScripts' => [],
+                'preScripts' => [],
+                'quota' => - 1,
+                'rsyncLongArgs' => '',
+                'rsyncShortArgs' => '',
+                'sshArgs' => '',
+                'url' => 'root@172.17.0.1'
+            ]
+        ]);
+        $iri = $this->findIriBy(Client::class, ['name' => 'client'.$timestamp.'0']);
+        
+        $httpClient->request('PUT', $iri, [
+            'json' => [
+                'description' => 'description updated',
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client',
+                'owner' => 1,
+                'quota' => - 1,
+                'url' => 'root@172.17.0.1'
+            ]
+        ]);
```

Comment:
You are not sending sshArgs, rsyncShortArgs etc...  and this is a PUT not a PATCH. Is this allowed?

It will delete/empty non provided properties or will it just leave them as they are? 

If it is the first one well that may make sense in the context of a REST PUT call but is a bit unexpected I'd rather we changed the API so that for a PUT all fields are mandatory. That change is out of the scope of this pull request but please file a bug/enhacement request so we don't lost track of this. And change this implementation so that when we make that change the test still works properly.

If it is the second one then this is not a proper REST PUT, it would be a PATCH. Please make sure that PUTs behave like PUTs should and not like PATCHes.

Apart from that I think that you should exercise changes in all fields of the client, not just some of them.